src/function/list/list_agg_function.cpp

```diff
@@ -0,0 +1,88 @@
+#include "common/exception/binder.h"
+#include "function/list/functions/list_product_function.h"
+#include "function/list/functions/list_sum_function.h"
+#include "function/list/vector_list_functions.h"
+#include "function/scalar_function.h"
+
+using namespace kuzu::common;
+
+namespace kuzu {
+namespace function {
+
+template<typename OPERATION>
+static std::unique_ptr<FunctionBindData> bindFuncListAggr(
+    const binder::expression_vector& arguments, Function* function) {
+    auto scalarFunction = ku_dynamic_cast<Function*, ScalarFunction*>(function);
+    auto resultType = ListType::getChildType(&arguments[0]->dataType);
+    switch (resultType->getLogicalTypeID()) {
+    case LogicalTypeID::SERIAL:
+    case LogicalTypeID::INT64: {
+        scalarFunction->execFunc =
```

Comment:
Not related to this PR, i think we can write another Typeutils::visit() function for logicalTypes, so we don't have to use such long switch/case statement.
Instead:
TypeUtils::visit(
                resultType->getLogicalTypeID(),
                [&]<typename T>(
                    T) {scalarFunction->execFunc =
            ScalarFunction::UnaryExecNestedTypeFunction<list_entry_t, T, OPERATION>;});